pallenec

```diff
@@ -70,21 +76,27 @@ local function pretty_print_ir(filename)
     io.stdout:write(print_ir(module))
 end
 
-local flags = {}
-if args.emit_c    then table.insert(flags, "--emit-c") end
-if args.emit_asm  then table.insert(flags, "--emit-asm") end
-if args.compile_c then table.insert(flags, "--compile-c") end
-if args.dump      then table.insert(flags, "--dump") end
-if args.print_ir  then table.insert(flags, "--print-ir") end
-if args.emit_lua  then table.insert(flags, "--emit-lua") end
-
-if #flags >= 2 then
-    local conflicting = table.concat(flags, " and ")
-    pallenec_abort("flags %s are mutually exclusive", conflicting)
+local flags = 0
+if args.emit_c    then flags = flags + 1 end
+if args.emit_asm  then flags = flags + 1 end
+if args.compile_c then flags = flags + 1 end
+if args.dump      then flags = flags + 1 end
+if args.print_ir  then flags = flags + 1 end
+if args.emit_lua  then flags = flags + 1 end
+
+local opt_passes = {}
+if args.optimizations then
+    for _, opt in ipairs(args.optimizations) do
+        opt_passes[opt] = true
+    end
+else
+    for _, opt in ipairs(all_opt_passes) do
+        opt_passes[opt] = true
+    end
 end
 
-if     #flags == 0    then compile("pln", "so")
-elseif args.emit_c    then compile("pln", "c")
+if     flags == 0     then compile("pln", "so", opt_passes)
```

Comment:
If we put this in the `else`, would this let us get rid of the `flags` variable?

Reply:
I thought about doing this, but I was afraid to lose the `error("impossible")`. If that is not a problem, I'll get rid of the `flag` variable.

Reply:
I think it is OK. I think we should have been using an `else` all along.